src/device-api/v1.ts

```diff
@@ -46,84 +49,50 @@ router.post('/v1/restart', (req: AuthorizedRequest, res, next) => {
 		.catch(next);
 });
 
-const v1StopOrStart = (
-	req: AuthorizedRequest,
-	res: express.Response,
-	next: express.NextFunction,
-	action: 'start' | 'stop',
-) => {
-	const appId = checkInt(req.params.appId);
-	const force = checkTruthy(req.body.force);
-	if (appId == null) {
-		return res.status(400).send('Missing app id');
-	}
-
-	return Promise.all([applicationManager.getCurrentApps(), getApp(appId)])
-		.then(([apps, targetApp]) => {
-			if (apps[appId] == null) {
-				return res.status(400).send('App not found');
-			}
-			const app = apps[appId];
-			let service = app.services[0];
-			if (service == null) {
-				return res.status(400).send('No services on app');
-			}
-			if (app.services.length > 1) {
-				return res
-					.status(400)
-					.send('Some v1 endpoints are only allowed on single-container apps');
-			}
-
-			// check that the request is scoped to cover this application
-			if (!req.auth.isScoped({ apps: [app.appId] })) {
-				return res.status(401).send('Unauthorized');
-			}
-
-			// Get the service from the target state (as we do in v2)
-			// TODO: what if we want to start a service belonging to the current app?
-			const targetService = _.find(targetApp.services, {
-				serviceName: service.serviceName,
-			});
+// QUESTION: This function is quite similar to v2's handleServiceAction
+// with the exception of some response messages being different and the presence
+// of the event tracker. Should we remove these differences and just use one function
+// between v1 & v2?
```

Comment:
If there is a way to reuse some of that code and just change the inputs/outputs to match the API specification, then I think that's a worthy improvement. Regarding the event tracker events, those are not really part of the public interface so we may add/remove them as necessary.

Reply:
Ended up not doing this -- while the logic was the same, there are too many differences in responses sent -- v2 service actions send JSON responses, and has a code path not in v1 service actions that handles when some fields aren't present in the request body. I don't see a way of doing it that isn't overly complex.